Update Windows EC2 instance storage-related settings

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR does the following:

- Bump the default EBS volume for Windows EC2 instances from `80` to `200`
- Update the Windows EC2 userdata to expand the main OS partition if it detects additional partitions
- Update the userdata script to run `always`
- Update to make sure that the script is using PowerShell style variable names
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

The default EBS size is increased because the base OS+installed utilities essentially fills the current allocation of `80`. Increasing to `200` should leave plenty of room for operators to perform work. The changes to userdata are primarily focused on resolving https://github.com/cisagov/cool-system-internal/issues/50, but they will require a new AMI that supports this change.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. Using this in my staging environment and Windows instances load with a 200GB OS partition.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] A supporting Windows AMI must exists in both the Production and Staging environments

mcdonnnj committed:
Increase the volume size for Windows instances

mcdonnnj committed:
Add partition extension to the Windows userdata

Due to the current image configuration we cannot rely on the built-in AWS
functionality to expand the partition for the OS automatically.

mcdonnnj committed:
Set Windows userdata to always run

This allows the userdata to support volume expansion after initial launch as
well as re-adding the Samba share in case it is missing for some reason.

mcdonnnj committed:
Adjust Windows userdata to make section idempotent

Make the portion of userdata that extends the OS partition idempotent.

mcdonnnj committed:
Adjust Windows userdata variable casing

Use PowerShell style variable casing for all variables in the script.

dav3r commented on ec2launch/windows-setup.tpl.yml:
```diff
@@ -1,23 +1,27 @@
         runAs: localSystem
         content: |-
-          $samba_server = ("${samba_server_input}" -split ",")[0]
+          $DiskPath = (Get-Partition -DriveLetter C).DiskPath
+          if ((Get-Partition -DiskId $DiskPath).Count -gt 2) {
+            "select disk 0`nselect partition 2`nextend`n" | diskpart.exe
```
Might be worth some additional comments here about magic drive letter `C` and magic partition `2`, unless you want to make those input variables.  Even then, there should probably still be a comment.

jsf9k replied:
I agree with @dav3r.

mcdonnnj replied:
I agree on the partition number, but `C` is the standard OS partition drive letter for Windows.

dav3r replied:
I get that, but even so, it doesn't hurt to mention in a comment that:
```
# C is the standard OS partition drive letter for Windows
```

mcdonnnj replied:
Well I was more referencing that we use a lot of Linux paths without explanation x)

dav3r replied:
I don't know if that's exactly analogous, but I leave it up to you unless @jsf9k feels strongly about this.

mcdonnnj replied:
I don't have any issue adding a comment explaining. My point was more it's worth commenting because Windows-related knowledge is not strongly in the domain knowledge of most of our team, but we use a lot of Linux-specific things without comment under the assumption that _it is_ in the domain knowledge of our team.

dav3r replied:
That's fair and I'm all for us doing a better job of adding comments for similar types of Linux thangs.

mcdonnnj replied:
I actually changed the `Get-Partition` call to be consistent with the commands we are piping to `diskpart.exe` in 1ccee97. I did add comments to overall explain things in the Windows instance userdata in 277ec3c.

dav3r replied:
Those comments look great to me - thanks!

dav3r reviewed (approved):
LGTM, but I did have one small bit of feedback. ⚙️

jsf9k reviewed (approved):
Approved, but please see @dav3r's request fro a comment explaining some magic numbers.

mcdonnnj committed:
Adjust Get-Partition usage in Windows userdata

Switch to using `-DiskNumber 0` instead of `-DriveLetter C` to get partition
information about the system disk. Using DiskNumber mirrors the commands we are feeding to
the `diskpart.exe` utility for consistency.

mcdonnnj committed:
Improve commenting in Windows userdata

Add comments to explain everything being done in the Windows instance userdata.
This will help when we look at it in three months and start to question the why
of what is being done.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
Co-Authored-By: Shane Frasier <maverick@maverickdolphin.com>

dav3r reviewed (approved):
Fantastic work! 🏖️
